Can't compile Codelite on Arch Linux

These code have been added in commit ca2a8b676f2a1cef8e9d7abd4ce13b8a4a6854b9. CMake error log.
````
codelite/Plugin/clDataViewListCtrl.cpp: In member function ‘bool clDataViewListCtrl::SendDataViewEvent(const wxEventType&, wxTreeEvent&)’:
codelite/Plugin/clDataViewListCtrl.cpp:145:67: error: no matching function for call to ‘wxDataViewEvent::wxDataViewEvent(const wxEventType&, wxDataViewListCtrl*, wxDataViewItem)’
     wxDataViewEvent e(type, &m_dummy, DV_ITEM(treeEvent.GetItem()));
                                                                   ^
In file included from codelite/Plugin/clDataViewListCtrl.h:6,
                 from codelite/Plugin/clDataViewListCtrl.cpp:1:
/usr/include/wx-3.0/wx/dataview.h:769:5: note: candidate: ‘wxDataViewEvent::wxDataViewEvent(const wxDataViewEvent&)’
     wxDataViewEvent(const wxDataViewEvent& event)
     ^~~~~~~~~~~~~~~
/usr/include/wx-3.0/wx/dataview.h:769:5: note:   candidate expects 1 argument, 3 provided
/usr/include/wx-3.0/wx/dataview.h:749:5: note: candidate: ‘wxDataViewEvent::wxDataViewEvent(wxEventType, int)’
     wxDataViewEvent(wxEventType commandType = wxEVT_NULL, int winid = 0)
     ^~~~~~~~~~~~~~~
/usr/include/wx-3.0/wx/dataview.h:749:5: note:   candidate expects 2 arguments, 3 provided
codelite/Plugin/clDataViewListCtrl.cpp: In member function ‘virtual void clDataViewListCtrl::SetSortFunction(const clSortFunc_t&)’:
codelite/Plugin/clDataViewListCtrl.cpp:338:10: error: ‘sort’ is not a member of ‘std’
     std::sort(children.begin(), children.end(), CompareFunc);
          ^~~~
codelite/Plugin/clDataViewListCtrl.cpp:338:10: note: suggested alternative: ‘sqrt’
     std::sort(children.begin(), children.end(), CompareFunc);
          ^~~~
          sqrt
make[2]: *** [Plugin/CMakeFiles/plugin.dir/build.make:1376: Plugin/CMakeFiles/plugin.dir/clDataViewListCtrl.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:314: Plugin/CMakeFiles/plugin.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
````

--- a/Plugin/clDataViewListCtrl.cpp
+++ b/Plugin/clDataViewListCtrl.cpp
@@ -147,7 +147,7 @@ bool clDataViewListCtrl::SendDataViewEvent(const wxEventType& type, wxTreeEvent&
 #if wxCHECK_VERSION(3, 1, 0)
     wxDataViewEvent e(type, &m_dummy, DV_ITEM(treeEvent.GetItem()));
 #else
-    wxDataViewEvent e(type));
+    wxDataViewEvent e(type);
     e.SetItem(DV_ITEM(treeEvent.GetItem()));
 #endif
     e.SetEventObject(this);

--- a/Plugin/clHeaderBar.cpp
+++ b/Plugin/clHeaderBar.cpp
@@ -93,7 +93,15 @@ void clHeaderBar::Render(wxDC& dc, const clColours& colours)
     _colours.SetBgColour(_colours.GetHeaderBgColour());
 
     bool useNativeHeader = (m_flags & kHeaderNative);
-    if(useNativeHeader) { wxRendererNative::Get().DrawHeaderButton(m_parent, dc, rect, wxCONTROL_NONE); }
+    if(useNativeHeader) {
+        #if wxCHECK_VERSION(3, 1, 0)
+            wxRendererNative::Get().DrawHeaderButton(m_parent, dc, rect, wxCONTROL_NONE);
+        #else
+            // Fourth argument is optional flags (none by default)
+            wxRendererNative::Get().DrawHeaderButton(m_parent, dc, rect);
+        #endif
+
+    }
 
     // Set the DC origin to reflect the h-scrollbar
     clControlWithItems* parent = dynamic_cast<clControlWithItems*>(GetParent());

--- a/Plugin/clHeaderItem.cpp
+++ b/Plugin/clHeaderItem.cpp
@@ -22,8 +22,12 @@ void clHeaderItem::Render(wxDC& dc, const clColours& colours, int flags)
     int textY = m_rect.GetY() + (m_rect.GetHeight() - textSize.GetHeight()) / 2;
 
     if(flags & kHeaderNative) {
-        wxRendererNative::Get().DrawHeaderButton(m_parent, dc, m_rect, wxCONTROL_NONE);
-
+        #if wxCHECK_VERSION(3, 1, 0)
+            wxRendererNative::Get().DrawHeaderButton(m_parent, dc, m_rect, wxCONTROL_NONE);
+        #else
+            // Fourth argument is optional flags (none by default)
+            wxRendererNative::Get().DrawHeaderButton(m_parent, dc, m_rect);
+        #endif
     } else {
         dc.SetBrush(colours.GetHeaderBgColour());
         dc.SetPen(colours.GetHeaderBgColour());

--- a/Plugin/drawingutils.cpp
+++ b/Plugin/drawingutils.cpp
@@ -678,7 +678,13 @@ void DrawingUtils::DrawNativeChoice(wxWindow* win, wxDC& dc, const wxRect& rect,
 #if 1
     // Windows & OSX
     wxRect choiceRect = rect;
-    wxRendererNative::Get().DrawChoice(win, dc, rect, wxCONTROL_NONE);
+    #if wxCHECK_VERSION(3, 1, 0)
+        wxRendererNative::Get().DrawChoice(win, dc, rect, wxCONTROL_NONE);
+    #else
+        // Fourth argument is optional flags (none by default)
+        wxRendererNative::Get().DrawChoice(win, dc, rect);
+    #endif
+
 #else
     // GTK
     wxColour face_light = wxSystemSettings::GetColour(wxSYS_COLOUR_3DFACE);

